article_maker.cc

```diff
@@ -376,9 +374,7 @@ sptr< Dictionary::DataRequest > ArticleMaker::makeNotFoundTextFor(
 
   sptr< Dictionary::DataRequestInstant > r = std::make_shared<Dictionary::DataRequestInstant>( true );
 
-  r->getData().resize( result.size() );
-  memcpy( &( r->getData().front() ), result.data(), result.size() );
-
```

Comment:
These lines can also be replaced with
```
std::copy(result.begin(),result.end(),r->getData().data());
```

Since the `result` is no longer muted by anyone else, `std::move` should be better.

Reply:
data continues
to get larger,using string maybe involve too much string concatenate which is not efficient

Reply:
too specific.

only on the first data , can done like this.